File: py/merge-two-binary-trees.py
class Solution(object):
    def mergeTrees(self, t1, t2):
        """
        :type t1: TreeNode
        :type t2: TreeNode
        :rtype: TreeNode
        """
        if t1 is None:
            return t2
        elif t2 is None:
            return t1
        else:
            ret = TreeNode(t1.val + t2.val)
            ret.left = self.mergeTrees(t1.left, t2.left)
            ret.right = self.mergeTrees(t1.right, t2.right)
            return ret

File: go/merge-two-binary-trees.go
package leetcode

func mergeTrees(t1 *TreeNode, t2 *TreeNode) *TreeNode {
	if t1 == nil && t2 == nil {
		return nil
	}
	if t1 != nil && t2 != nil {
		ans := &TreeNode{}
		ans.Val = t1.Val + t2.Val
		ans.Left = mergeTrees(t1.Left, t2.Left)
		ans.Right = mergeTrees(t1.Right, t2.Right)
		return ans
	} else if t1 != nil {
		return t1
	} else if t2 != nil {
		return t2
	}
	return nil
}
